scala-package/core/src/main/scala/org/apache/mxnet/io/NDArrayIter.scala

```diff
@@ -42,27 +43,40 @@ import scala.collection.immutable.ListMap
 class NDArrayIter(data: IndexedSeq[(String, NDArray)],
                   label: IndexedSeq[(String, NDArray)],
                   private val dataBatchSize: Int, shuffle: Boolean,
-                  lastBatchHandle: String) extends DataIter {
-
+                  lastBatchHandle: String,
+                  dataDType: DType, labelDType: DType,
+                  dataLayout: String, labelLayout: String) extends DataIter {
+
+  // scalastyle:off
```

Comment:
why?

Reply:
I finally figure out the scala style issue. Scala does not allow user to define a function with more than a certain numbers of args...